src/main.rs

```diff
@@ -13,7 +14,21 @@ mod sc_generation;
 fn main() {
```

Comment:
doesn't modifying fn main broke the gas calibration for legacy abis?
(I dunno if it's a major point)

Reply:
Mhh it's true that I didn't keep track of what is breaking. The main should not be a problem, but for example the generation.rs file only takes into account new abis.
I could pass around a CLI argument to switch the behaviour if it's needed.

Or maybe we could tag the current main so we can easily revert to it.

What do you think?

Reply:
Yes I've seen other changes that are also breaking the compatibility with the legacy abis.
As we are about to remove the legacy abis, your idea of a tag looks enought to me.
I'd ask the point of view of @AurelienFT on this point^^

Reply:
We don't need to think about the old abi anymore they have been calibrated long time ago and we will not calibrate it again